js/store.js

```diff
@@ -113,11 +203,49 @@ const store = {
 
     this._write(websites);
 
-    if (newSite) {
+    if (newSite && data['isVisible']) {
```

Comment:
Yes you're right; making that key always present should allow this to work (see previous comment above).